### PR TITLE
112 new class format

### LINKED ISSUE #112: New Class Format for Management API v2
### Motivation

Related to https://github.com/Kentico/kontent-management-sdk-net/pull/67

As a part of the upgrade, we decided to:
* define types for model's elements
* decorate every element for the model with JsonProperty("<elementcodename>")
* decorate every element with custom attribute KontentElementId(<elementid>)

### Proposed solution

Example of the mode would looks like that:

> **The name of the attributes might be a subject of change**

* The namespaces are illustrative, wee need to add the ones we need
* Namespace for KontentElementId Attribute will be probably `Kentico.Kontent.Management.Modules.ModelBuilders`

```csharp
using System;
using Kentico.Kontent.Management.Models.Assets;
using Kentico.Kontent.Management.Models.Items;
using Kentico.Kontent.Management.Models.Items.Elements;
using Kentico.Kontent.Management.Modules.ModelBuilders;
using Newtonsoft.Json;

namespace KenticoKontentModels
{
    public partial class MyContentType
    {
        [JsonProperty("text")]
        [KontentElementId("77108990-3c30-5ffb-8dcd-8eb85fc52cb1")]
        public TextElement Text { get; set; }
    }
}
```

### Additional context

Don't forget about the docs, but wait for the initial implementation review so that we don't need to rewrite documentation multiple times.

### PR DESCRIPTION
**Pull request by Sevitas**
### Motivation

Fixes #112 

### Checklist

- [x] Code follows coding conventions held in this repo
- [x] Automated tests have been added
- [x] Tests are passing
- [ ] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

Generate models for DAPI/MAPI
to generate mapi model you need to provide a MAPI token in command line args with _-k_ arg

### CODE REVIEW

File: src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs [hunk 1]
```diff
@@ -34,81 +42,81 @@ public async Task<int> RunAsync()
...
+            WriteToOutputProvider(typeProviderCode, TypeProviderCodeGenerator.ClassName, true);
         }
 
-        internal async Task<IEnumerable<ClassCodeGenerator>> GetClassCodeGenerators(bool structuredModel = false)
+        internal async Task<ICollection<ClassCodeGenerator>> GetClassCodeGenerators()
         {
-            IEnumerable<IContentType> contentTypes = (await _client.GetTypesAsync()).Types;
+            var deliveryTypes = (await _client.GetTypesAsync()).Types;
+            var managementTypes = await _managementClient.GetAllContentTypesAsync(_options);
+            var managementSnippets = await _managementClient.GetAllSnippetsAsync(_options);
```
Comment (Simply007) [inline src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs:87]: All of the data are required for delivery model generations as well?

File: test/Kentico.Kontent.ModelGenerator.Tests/ClassCodeGeneratorFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using JetBrains.Annotations;
```
Comment (Simply007) [inline test/Kentico.Kontent.ModelGenerator.Tests/ClassCodeGeneratorFactoryTests.cs:2]: I would remove Jetbrains anotations.

**Review by Simply007 — CHANGES_REQUESTED** [review]
Just two minor issues.

I might split the code into folders. If you think we should split the folder categorization into separate issue/pull requests, feel free to do so.

File: src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs [hunk 3]
```diff
@@ -34,81 +47,87 @@ public async Task<int> RunAsync()
...
+        internal async Task<ICollection<ClassCodeGenerator>> GetClassCodeGenerators()
         {
-            IEnumerable<IContentType> contentTypes = (await _client.GetTypesAsync()).Types;
+            var deliveryTypes = (await _client.GetTypesAsync()).Types;
+            IList<ContentTypeModel> managementTypes = null;
+            IList<SnippetModel> managementSnippets = null;
+
+            if (_options.ContentManagementApi)
+            {
+                managementTypes = await _managementClient.GetAllContentTypesAsync(_options);
```
Comment (Simply007) [inline src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs:96]: Might be interesting to use Lazy - what do you think @Sevitas?

**Review by Simply007 — COMMENTED** [review]
One last comment + other findings submitted separately #124.

File: src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs (hunk 3 above)
Comment (Sevitas) [reply src/Kentico.Kontent.ModelGenerator.Core/CodeGenerator.cs:96]: We have decided not to go with a Lazy thus snippets are used in other classes as well. Passing list of objects looks more clean than passing Lazy object.

**Review by Simply007 — APPROVED** [review]
Looks goog now, thanks!

**Pull request merged**